lib/more_core_extensions/core_ext/string/decimal_suffix.rb

```diff
@@ -0,0 +1,19 @@
+module MoreCoreExtensions
+  module DECIMAL_SI
+    # made to handle https://github.com/kubernetes/kubernetes/blob/4def5add114b651c26fe576b7315f8029bfce46a/vendor/github.com/appc/spec/schema/types/resource/quantity.go#L49
+    DECIMAL_SUFFIXES = %w(k M G T P E).freeze
```

Comment:
a hash `{"m" => 1e-3, "k" => 1e3, "M" => 1e6, ...}` might make the logic simpler.